Moving target: upcoming version

lumiscosity opened the pull request:
Based off of #58 with the code review comments applied. This is the modified copy mentioned in https://github.com/sisby-folk/antique-atlas/pull/246. Confirmed working on 1.21.2, 1.21.3 and 1.21.4-pre1.

Businessman381 committed:
Ported to 1.21.2

lumiscosity committed:
address code review concerns

sleepingdragoninn commented on src/main/resources/fabric.mod.json:
```diff
@@ -16,15 +16,15 @@
-	"accessWidener": "${modId}.accesswidener",
+	"icon": "assets/surveyor/icon.png",
+	"accessWidener": "surveyor.accesswidener",
 	"mixins": [
-		"${modId}.mixins.json"
+		"surveyor.mixins.json"
 	],
 	"depends": {
-		"minecraft": ">=${mc}",
-		"fabricloader": ">=${fl}",
-		"fabric-api": ">=${fapi}"
+		"minecraft": ">=1.21.2",
+		"fabricloader": ">=0.16.7",
+		"fabric-api": ">=0.106.1+1.21.2"
 	},
```
nopety! this is all controlled by gradle.properties. every change here can just be straight removed, it's automatic.

sleepingdragoninn commented on src/main/resources/META-INF/neoforge.mods.toml:
```diff
@@ -15,28 +15,28 @@ logoFile = "assets/${modId}/icon.png"
 ordering = "NONE"
 side = "BOTH"
 
-[[dependencies.${ modId }]]
```
why's this a thing?

lumiscosity replied:
i responded to the comment on the initial issue about this, i initially kept this tweak because that's how the default neoforge mod template does it, and intellij idea throws a warning if the spaces are present.

since it works fine anyways, rolled back the change for now.

sleepingdragoninn replied:
I'll put it in 1.21 instead and merge it over!

sleepingdragoninn commented on gradle.properties:
```diff
@@ -19,7 +19,7 @@ license=LGPL-3.0-or-later
 baseVersion=0.6.25
 # Branch Metadata
-branch=1.21
-tagBranch=1.20
```
not quite! the tag branch is the branch we develop features on - that stays at `1.20`.
Additionally, the branch for this is `1.22`

sleepingdragoninn commented on src/main/java/folk/sisby/surveyor/terrain/ChunkSummary.java:
```diff
@@ -76,7 +76,7 @@ public ChunkSummary(World world, WorldChunk chunk, int[] layerHeights, RegistryP
 						} else { // Blocks Movement or Has Non-Water Fluid.
 							if (foundFloor == null) {
 								if (carpetPos.getY() == y + 1) {
-									foundFloor = new LayerSummary.FloorSummary(carpetPos.getY(), biomePalette.findOrAdd(section.getBiomeEntry(x, carpetPos.getY(), z, world.getBottomY(), world.getTopY()).value()), blockPalette.findOrAdd(carpetBlock), world.getLightLevel(LightType.BLOCK, carpetPos), waterDepth, waterDepth == 0 ? 0 : world.getLightLevel(LightType.BLOCK, pos.up().up(waterDepth)));
```
for completeness, it's always good to confirm a surveyor update with a hoofprint update as well, just because atlas hides a lot of detail that could become wrong when touching the scanner. I think you're fine here because these are just renames, i can only assume the impl is the same.

lumiscosity replied:
i just got an initial port going but i'm encountering the same rendering issues as in aa4. though considering the fact that hoofprint has way less parts i think i could reasonably try to figure it out, and maybe the solution will work for both

sleepingdragoninn reviewed (approved):
Thanks for the assist! still some things left over though

lumiscosity committed:
address code review concerns, part 2

lumiscosity committed:
move target to latest snapshot

not quite done yet! the two unnamed methods in MixinServerWorld are gone and i'll have to track down what they were renamed to. other than that, it's mostly yarn renames.

sleepingdragoninn commented:
hi again! we'll be looking to pick this up and try target a 1.21.7 release pretty soon

lumiscosity commented:
got it, i'll try to get it up to speed with the main branch today, hopefully they fixed mixins not being able to target closures

edit: looks like i was wrong about the closure thing, i've moved my dev env to 1.21.5 proper and the closures are indeed at new methods now due to the introduction of a POI manager in vanilla

lumiscosity commented:
wait. you already rebased it. well, there goes my free morning. sorry for making a mess!

lumiscosity commented:
nevermind, i'm struggling with the rebase and i'm not sure how to fix it...

lumiscosity committed:
finish 1.21.5 port

loom 1.7 can no longer be used, so this commit also updates it to 1.10.5, bumping the required gradle version to 8.12.

not sure if all the .get() calls are good code hygiene. my intuition says these need some backup checks.... whatever, i'd like to get it running first and proper second

i was hoping this would work on 1.21.6 but unfortunately there's more changes in there and a mixin's failing. bumping to 1.21.6 next, then i'll rebase

lumiscosity committed:
most of a 1.21.6 port

writeview and readview are cool but unfortunately we need to read in a mixin to readview. this creates a cascade of utter bs that i patched over until it built, except it very much does not work. i'm not qualified to rewrite the entire player/server summary api to match this so this is now a sisby problem

lumiscosity commented:
1.21.5 port is functional! probably! with no tests or hoofprint to go off of i take "lets me log in and run around" as a win

tried (builds but crashes on world join) porting to 1.21.6 which should be identical to 1.21.7 in the things that matter - that being the fact that the `NbtCompound` on the player is now exclusively accesed through `ReadView` and `WriteView`, so the entire api there needs a ground up rewrite i'm not qualified to make. 

no rebasing has happened either.

i've reached the point where i don't know how to continue this so the rest of the work is up to you

sleepingdragoninn commented:
thanks maple! I'll give it a whirl from here
